### PR TITLE
Do not return deleted files by default in `get_staged_files`

### PR DESCRIPTION
**Pull request by FlorentClarret**
### What does this PR do?

Do not return deleted files by default in `get_staged_files`

### Motivation

We usually do not need to check things on deleted files

### CODE REVIEW

File: tasks/libs/common/git.py
Comment (pgimalac) [inline tasks/libs/common/git.py:1]: I'm guessing that makes `govet` not run on packages where we only deleted files ?

File: tasks/libs/common/git.py [hunk 1]
```diff
@@ -1,8 +1,25 @@
...
-    return ctx.run(f"git diff --name-only --staged {commit}", hide=True).stdout.strip().splitlines()
+
+    files = ctx.run(f"git diff --name-only --staged {commit}", hide=True).stdout.strip().splitlines()
+
+    if include_deleted_files:
+        yield from files
+    else:
+        for file in files:
+            if os.path.isfile(file):
+                yield file
```
Comment (chouetz) [inline tasks/libs/common/git.py:22]:
```suggestion
        yield from [file for file in files if os.path.isfile(file)]
```
I don't know if you can write it this way

File: tasks/libs/common/git.py [hunk 2]
```diff
@@ -1,8 +1,25 @@
...
+from __future__ import annotations
+
+import os.path
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable
+
+
+def get_staged_files(ctx, commit="HEAD", include_deleted_files=False) -> Iterable[str]:
```
Comment (chouetz) [inline tasks/libs/common/git.py:10]: Should we include by default?

**Review by chouetz — APPROVED** [review]

File: tasks/libs/common/git.py (hunk 1 above)
Comment (FlorentClarret) [reply tasks/libs/common/git.py:22]: It would work but it would create a temporary list which defeats the purpose of returning an iterator

**Pull request merged**